package.json

```diff
@@ -33,7 +35,8 @@
     "prepare": "yarn simple-git-hooks",
     "dev:back": "yarn workspace confiture-rest-api run start:dev",
     "dev:front": "yarn workspace confiture-web-app run dev",
-    "dev": "yarn dev:back & yarn dev:front"
+    "dev": "yarn dev:back & yarn dev:front",
+    "g:show-all-ts-errors": "cd $INIT_CWD && tsc --noEmit"
```

Comment:
C'est pour faire quoi cette commande ? Notamment `$INIT_CWD`.

Reply:
Elle est utilisée depuis le workspace `confiture-rest-api` voir `"show-all-ts-errors"` dans `confiture-rest-api/package.json`

Avant le script dans `confiture-rest-api` était simplement `"tsc --noEmit"`. Mais ça ne fonctionne pas si `typescript` n’est pas défini comme "dev Dependency" dans `confiture-rest-api`. L’idée c’est de définir `typescript` une seule fois à la racine du projet (comme ça pas de risque de versions différentes entres workspaces)

`cd $INIT_CWD` c’est pour aller dans le répertoire depuis lequel ce script global est appelé.

Voir : [How to share scripts between workspaces?](https://yarnpkg.com/getting-started/qa#how-to-share-scripts-between-workspaces)